plan/plans/perf.go

```diff
@@ -0,0 +1,1011 @@
+// Copyright 2015 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plans
+
+import (
+	"strings"
+
+	"github.com/juju/errors"
+	"github.com/pingcap/tidb/column"
+	"github.com/pingcap/tidb/context"
+	"github.com/pingcap/tidb/expression"
+	"github.com/pingcap/tidb/field"
+	"github.com/pingcap/tidb/model"
+	"github.com/pingcap/tidb/mysql"
+	"github.com/pingcap/tidb/perfschema"
+	"github.com/pingcap/tidb/plan"
+	"github.com/pingcap/tidb/util/charset"
+	"github.com/pingcap/tidb/util/format"
+	"github.com/pingcap/tidb/util/types"
+)
+
+var _ = (*PerfSchemaPlan)(nil)
+
+// PerfSchemaPlan handles performance_schema query, simulates the behavior of
+// MySQL.
+type PerfSchemaPlan struct {
+	TableName string
+	rows      []*plan.Row
+	cursor    int
+}
+
+var (
+	setupActorsFields            []*field.ResultField
```

Comment:
@qiuyesuifeng Do you mean the variables? I know alphabetical order is good, but this is the same as the order in MySQL's document, do you insist your opinion?